kotlintest-runner/kotlintest-runner-junit5/src/main/kotlin/io/kotlintest/runner/junit5/JUnitTestRunnerListener.kt

```diff
@@ -193,9 +187,11 @@ class JUnitTestRunnerListener(private val listener: EngineExecutionListener,
       logger.error("Spec descriptor cannot be null $description")
       throw RuntimeException("Spec descriptor cannot be null")
     } else {
-      val result = when (t) {
-        null -> TestExecutionResult.successful()
-        else -> TestExecutionResult.failed(t)
+      val firstTestError = results.find { it.result.error != null }?.result?.error
```

Comment:
I am pretty sure that the JUnit specs say that a parent must pass even if it's children fail.
This is why I do not search the children for a failed test.

Reply:
Perhaps. Let me see what I can do.

Reply:
The thing is, if we don't fail the parent, the test case won't ever be created

Reply:
perhaps we can add the dummy failure like we do with our runner